### PR TITLE
Add date created to output

### PR DESCRIPTION
**Pull request by pmdredd**
This PR aims to solve issue #8 

I have added a new attribute to the IssueModel.php which represents the created_date of a GitHub issue, implemented as a php DateTime class. Initially I displayed the value of this attribute to the output of RandomIssueCommand.execute() however in response to PR #10 by DenverCoder1 I have removed my changes to that class to prevent any merge conflicts. After that branch is merged (or rejected) I can push another commit to this PR that adds outputting the created_date before merging. From the wording of issue #8 only the date should be shown, not the full DateTime.

I reviewed GitHub's documentation available [here](https://docs.github.com/en/free-pro-team@latest/rest/reference/search#search-issues-and-pull-requests) and found that their API returns the "created_at", "updated_at" and "closed_at" datetimes. I assume that for now this tool only wants to pick issues that are not closed (which excludes showing the "closed_at" date), and I made the decision to only include the "created_at" date as I have found the "updated_at" date to be less important in my experience.

Note: An alternative implementation is to instantiate the new DateTime object in the constructor of IssueModel rather than every time getCreatedAt() is called. If we were to re-use the IssueModel multiple times then I think it would be better to instantiate in the constructor.

### All Submissions:

* [x] Does your PR have the correct target branch?
* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/<User>/<Repository>/pulls) for the same update/change?

### New Feature Submissions:

1. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: src/Model/IssueModel.php
Comment (icanhazstring) [inline src/Model/IssueModel.php]: Using `DateTime` inherits the problem of a mutable state of this object.
Please use `DateTimeImmutable` for this purpose.

File: src/Model/IssueModel.php
Comment (icanhazstring) [inline src/Model/IssueModel.php]: If you know the format of `created_at` the JMS\Serializer can automatically parse the date string into an actual object.

See the documentation here: https://jmsyst.com/libs/serializer/master/reference/annotations#type
For this (as mentioned above) use `DateTimeImmutable<...>` as the type.

File: src/Model/IssueModel.php
Comment (icanhazstring) [inline src/Model/IssueModel.php]: The return type should be changed accordingly

File: src/Model/IssueModel.php
Comment (icanhazstring) [inline src/Model/IssueModel.php]: This is not needed as mentioned above. The JMS\Serializer already transforms the incoming date string from `created_at` into the proper type.

**Review by icanhazstring — CHANGES_REQUESTED** [review]
Thanks for your contribution 👍 
There are some things that need attending to.

Also: could you update your branch with latest master so I could merge this after you done the required changes.

**Comment by icanhazstring** [comment]
There was already a claim for #8 from another person. PR #12 and this one are aiming for the same issue but I will merge both of them. As they are both valid PRs 👍

**Comment by pmdredd** [comment]
I've updated my branch in line with the most recent commits to master and altered the code as per the PR comments. I've also added a basic ->format() method to ->getCreatedAt() to prevent the console complaining about the wrong type being supplied.

I'm more than happy to let @DenverCoder1 have a crack at formatting the date in a human-readable way (#14).

**Comment by icanhazstring** [comment]
Nice thank you. Looks good to me.
If @DenverCoder1 has added his changes we can merge it 👍

**Comment by DenverCoder1** [comment]
> I'm more than happy to let DenverCoder1 have a crack at formatting the date in a human-readable way (#14).

How about something like `->format('D, j M Y g:i A \U\T\C')`     
which will output the format: `Sun, 6 Sep 2020 5:59 PM UTC`

Probably don't want to make it too region specific (like m/d/yyyy is not as common as d/m/yyyy). Also, not sure about keeping the AM/PM.

Feel free to make the final decision.

**Comment by icanhazstring** [comment]
> How about something like ->format('D, j M Y g:i A \U\T\C')
which will output the format: Sun, 6 Sep 2020 5:59 PM UTC

That looks good to me 👍 
Also, I like the idea to keep it non-region-specific.

**Comment by DenverCoder1** [comment]
> If @DenverCoder1 has added his changes we can merge it 👍

Is this something I should do? If so I'm not sure how to add things to the PR (this is my first time).

**Comment by icanhazstring** [comment]
> Is this something I should do? If so I'm not sure how to add things to the PR (this is my first time).

You can use the github cli (https://github.com/cli/cli)
With this you can check out a certain PR from this repo

```
gh repo clone icanhazstring/random-issue-picker
gh pr checkout 11
// do changes
git commit ...
git push
```

**Comment by icanhazstring** [comment]
But @pmdredd needs to allow changes from you on this PR :)

https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by DenverCoder1** [comment]
> But @pmdredd needs to allow changes from you on this PR

Right, I was able to get most of the way through, but when I tried to Push it said Authentication failed (probably because no write access).

**Comment by pmdredd** [comment]
I've checked the box "Allow edits by maintainers" as per the docs you linked however I'm not sure if that will allow @DenverCoder1 to push to this PR (fork?). I've invited them to my forked repo as well.

**Comment by icanhazstring** [comment]
The invite should work I guess 👍

**Comment by DenverCoder1** [comment]
The invite worked. Should be good now.

**Review by icanhazstring — APPROVED** [review]
Thanks both of you. Merged 👍

**Pull request merged**
